repository/firestore.go

```diff
@@ -0,0 +1,97 @@
+package repository
+
+import (
+	"cloud.google.com/go/firestore"
+	"context"
+	"errors"
+	"github.com/gazebo-web/gz-go/v7/reflect"
+)
+
+var (
+	// ErrMethodNotImplemented is returned when a certain method is not implemented.
+	ErrMethodNotImplemented = errors.New("method not implemented")
```

Comment:
Consider moving this to the generic `repository.go` file so that it can be used by other implementations.